### PR TITLE
fix jenkins header & url

### PR DESCRIPTION
**Pull request by soulseen**
Signed-off-by: soulseen <sunzhu@yunify.com>

### CODE REVIEW

File: pkg/models/devops/devops.go [hunk 1]
```diff
@@ -678,19 +684,21 @@ func sendJenkinsRequest(baseUrl string, req *http.Request) ([]byte, error) {
...
 	if resp.StatusCode >= http.StatusBadRequest {
 		jkerr := new(JkError)
 		jkerr.Code = resp.StatusCode
 		jkerr.Message = http.StatusText(resp.StatusCode)
-		return nil, jkerr
+		return nil, nil, jkerr
 	}
 
-	return resBody, err
+	defer resp.Body.Close()
```
Comment (runzexia) [inline pkg/models/devops/devops.go]: if resp > 400, resp body not closed

File: pkg/apiserver/devops/devops.go [hunk 2]
```diff
@@ -119,11 +123,14 @@ func GetStepLog(req *restful.Request, resp *restful.Response) {
...
 	stepId := req.PathParameter("stepId")
 
-	res, err := devops.GetStepLog(projectName, pipelineName, runId, nodeId, stepId, req.Request)
+	res, header, err := devops.GetStepLog(projectName, pipelineName, runId, nodeId, stepId, req.Request)
 	if err != nil {
 		parseErr(err, resp)
 		return
 	}
+	for k, v := range header {
+		resp.AddHeader(k, v[0])
```
Comment (runzexia) [inline pkg/apiserver/devops/devops.go]: Add X-.* header?

File: pkg/apiserver/devops/devops.go [hunk 3]
```diff
@@ -104,11 +104,15 @@ func GetBranchStepLog(req *restful.Request, resp *restful.Response) {
...
 
-	res, err := devops.GetBranchStepLog(projectName, pipelineName, branchName, runId, nodeId, stepId, req.Request)
+	res, header, err := devops.GetBranchStepLog(projectName, pipelineName, branchName, runId, nodeId, stepId, req.Request)
+
 	if err != nil {
 		parseErr(err, resp)
 		return
 	}
+	for k, v := range header {
+		resp.AddHeader(k, v[0])
```
Comment (runzexia) [inline pkg/apiserver/devops/devops.go]: same

File: pkg/apiserver/devops/devops.go (hunk 2 above)
Comment (soulseen) [reply pkg/apiserver/devops/devops.go]: yes

**Pull request merged**